Sources/Files.swift

```diff
@@ -265,6 +284,24 @@ public class FileSystem {
         }
         
         /**
+         *  Copy this item to a new folder
+         *
+         *  - parameter folder: The folder that the item should be copy to
+         *
+         *  - throws: `FileSystem.Item.OperationError.copyFailed` if the item couldn't be copied
+         */
+        public func copy(to folder: Folder) throws {
+            let newPath = folder.path + name
+            
+            do {
+                try fileManager.copyItem(atPath: path, toPath: newPath)
+                path = newPath
```

Comment:
I don't think copying the item should modify its path, since the item being copied remains at the same location. Instead, I think this method should return a new item that represents the copy.

Reply:
We never return `FileSystem.Item` from any public API though, so you might want to implement it separately on `File` and `Folder` in order to return the correct type. Also, it should be marked as `@discardableResult` so that the user doesn't get a warning in case they're not consuming the returned object.

Reply:
Good remarks! And stupid mistakes. About two implementations - it will be code duplication. Did you think about generics of associated type for it?